### DISCUSSION TITLE
how do I parse a resource from input json file

### DISCUSSION THREAD

**Discussion by aggaa-netapp** [fabric8io/kubernetes-client#5504] [General]
If I have the resource (say Ingress) specified in a Json file. How do I create Ingress object from it? Thanks.

**Comment by rohanKanojia** [comment]
I think it should be like this:
https://github.com/fabric8io/kubernetes-client/blob/e4947f762697be50dee06944795b58386216e8b8/kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/NetworkingV1IngressTest.java#L50

```java
Ingress ing = client.network().ingresses().load(YourClass.class.getResourceAsStream("/test-ing.json")).get();
```

**Reply by shawkins** [reply]
On later releases use item() rather than get()

**Reply by aggaa-netapp** [reply]
Thanks.. but why is **client** needed for this. I would think that just the Ingress class should be enough to parse that resource from json file.

**Reply by shawkins** [reply]
The class is nearly enough.  In cases where none of the descendents are typed as KubernetesResource, then just using the class and an ObjectMapper is pretty close.  However there are quite a few customizations of an ObjectMapper that fabric8 users and we also handle the case of nested KubernetesResources - in recent releases this is via a class called KubernetesSerialization.  If you wish, you could directly use that instead of the KubernetesClient methods.

**Comment by aggaa-netapp** [comment]
And a follow up question. Does the mock server client support this load method as well?

**Reply by shawkins** [reply]
No, the mock server is typically not involved in the processing of load.

**Reply by aggaa-netapp** [reply]
Yes, I am seeing that mock server client is returning **null** object back (as against throwing exception) when I try to use the load() method on it.

**Reply by shawkins** [reply]
To clarify the mock server shouldn't be involved with the processing of the load method specifically, but it could be involved in the preceeding or following methods.  On newer releases when you do load(...).get() - the get is a fetch from the server.  In older releases it is just getting the client context item.  To make things clearer there is now the item() method instead.

**Reply by aggaa-netapp** [reply]
I see. Yes thats what was happening. It was trying to get/fetch from server and returning null. I changed to item() instead of get() and now it does return the resource object loaded from input json file. I am assuming you meant item() when you said getitem() above. I don't see a method called getitem()

**Reply by shawkins** [reply]
Yes, sorry - should have checked the source first.  Corrected the comment above as well.

**Reply by aggaa-netapp** [reply]
Thank you for your help!
